pyproject.toml

```diff
@@ -29,7 +29,7 @@ classifiers = [
     "Programming Language :: Python :: 3.12",
 ]
 dependencies = [
-    "apache-airflow>=2.3",
+    "apache-airflow>=2.3,<3.0",
```

Comment:
note that I'm pinning to `<3.0` until we support 3.0 (e.g. with https://github.com/astronomer/dag-factory/pull/413 and other necessary changes)

Reply:
It's fine without this because we only install explicit versions of airflow, and 3.0 isn't one of them. where I was running into issues was with local dev - installing the package also brings the latest version of airflow by default, and it doesn't work with 3.0